src/lava/networks/gradedvecnetwork.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (C) 2022-23 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import numpy as np
+from scipy.sparse import csr_matrix
+
+from lava.proc.graded.process import InvSqrt
+from lava.proc.graded.process import NormVecDelay
+from lava.proc.sparse.process import Sparse
+from lava.proc.dense.process import Dense
+from lava.proc.prodneuron.process import ProdNeuron
+from lava.proc.graded.process import GradedVec
+from lava.proc.lif.process import LIF
+from lava.proc.io import sink, source
+from lava.proc import embedded_io as eio
+
+from lava.magma.core.process.variable import Var
+from lava.magma.core.process.ports.ports import InPort, OutPort
+
+from .network import AlgebraicVector, AlgebraicMatrix
+
+
+class InputVec(AlgebraicVector):
+    """InputVec
+    Simple input vector. Adds algebraic syntax to RingBuffer
+
+    Parameters
+    ----------
+    vec: np.ndarray
+        NxM array of input values. Input will repeat every M steps.
+    loihi2=False: bool
+        Flag to create the adapters for loihi 2.
+    """
+
+    def __init__(self, vec, **kwargs):
+
+        self.loihi2 = kwargs.pop('loihi2', False)
+        self.shape = np.atleast_2d(vec).shape
+        self.exp = kwargs.pop('exp', 0)
+        print(self.shape)
+
+        # convert it to fixed point base
+        vec *= 2**self.exp
+
+        self.inport_plug = source.RingBuffer(data=np.atleast_2d(vec))
+
+        if self.loihi2:
+            self.inport_adapter = eio.spike.PyToNxAdapter(
+                shape=(self.shape[0],),
+                num_message_bits=24)
+            self.inport_plug.s_out.connect(self.inport_adapter.inp)
+            self.out_port = self.inport_adapter.out
+
+        else:
+            self.out_port = self.inport_plug.s_out
+
+    def __lshift__(self, other):
+        # maybe this could be done with a numpy array and call set_data?
+        return NotImplemented
+
+
+class OutputVec(AlgebraicVector):
+    """OutputVec
+    Records spike output. Adds algebraic syntax to RingBuffer
+
+    Parameters
+    ----------
+    shape=(1,): tuple(int)
+        shape of the output to record
+    buffer=1: int
+        length of the recording.
+        (buffer is overwritten if shorter than sim time).
+    loihi2=False: bool
+        Flag to create the adapters for loihi 2.
+    num_message_bits=24: int
+        size of output message. ("0" is for unary spike event).
+    """
+
+    def __init__(self, **kwargs):
+
+        self.shape = kwargs.pop('shape', (1,))
+        self.buffer = kwargs.pop('buffer', 1)
+        self.loihi2 = kwargs.pop('loihi2', False)
+        self.num_message_bits = kwargs.pop('num_message_bits', 24)
```

Comment:
See comments about docstrings, `__init__` signature and type hints I added for `InputVec`.